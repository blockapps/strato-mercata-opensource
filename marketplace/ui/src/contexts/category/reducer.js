import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
