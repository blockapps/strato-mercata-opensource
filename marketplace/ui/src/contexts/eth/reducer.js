import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchETHSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchETHSTAddressSuccessful:
      return {
        ...state,
        ethstAddress: action.payload,
      };
    case actionDescriptors.fetchETHSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.addHash:
      return {
        ...state,
        isAddingHash: true,
      };
    case actionDescriptors.addHashSuccessful:
      return {
        ...state,
        isAddingHash: false,
      };
    case actionDescriptors.addHashFailed:
      return {
        ...state,
        isAddingHash: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
