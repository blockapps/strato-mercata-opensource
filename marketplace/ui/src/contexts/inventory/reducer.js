import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createInventory:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createInventoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchInventory:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUser:
      return {
        ...state,
        isUserInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventoryForUserSuccessful:
      return {
        ...state,
        userInventories: action.payload.data,
        userInventoriesTotal: action.payload.count,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUserFailed:
      return {
        ...state,
        error: action.error,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearch:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySearchSuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearchFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.updateInventory:
      return {
        ...state,
        isinventoryUpdating: true,
      };
    case actionDescriptors.updateInventorySuccessful:
      return {
        ...state,
        inventoryUpdateObject: action.payload,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateInventoryFailed:
      return {
        ...state,
        error: action.error,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateSale:
      return {
        ...state,
        issaleUpdating: true,
      };
    case actionDescriptors.updateSaleSuccessful:
      return {
        ...state,
        saleUpdateObject: action.payload,
        issaleUpdating: false,
      };
    case actionDescriptors.updateSaleFailed:
      return {
        ...state,
        error: action.error,
        issaleUpdating: false,
      };
    case actionDescriptors.listInventory:
      return {
        ...state,
        isListing: true,
      };
    case actionDescriptors.listInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isListing: false,
      };
    case actionDescriptors.listInventoryFailed:
      return {
        ...state,
        error: action.error,
        isListing: false,
      };
    case actionDescriptors.unlistInventory:
      return {
        ...state,
        isUnlisting: true,
      };
    case actionDescriptors.unlistInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isUnlisting: false,
      };
    case actionDescriptors.unlistInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnlisting: false,
      };

     // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.stakeInventory:
      return {
        ...state,
        isStaking: true
      };
    case actionDescriptors.stakeInventorySuccessful:
      return {
        ...state,
        isStaking: false
      };
    case actionDescriptors.stakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isStaking: false
      };
    
    case actionDescriptors.unstakeInventory:
      return {
        ...state,
        isUnstaking: true
      };
    case actionDescriptors.unstakeInventorySuccessful:
      return {
        ...state,
        isUnstaking: false
      };
    case actionDescriptors.unstakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnstaking: false
      };
    
    case actionDescriptors.borrow:
      return {
        ...state,
        isBorrowing: true
      };
    case actionDescriptors.borrowSuccessful:
      return {
        ...state,
        isBorrowing: false
      };
    case actionDescriptors.borrowFailed:
      return {
        ...state,
        error: action.error,
        isBorrowing: false
      };
    
    case actionDescriptors.repay:
      return {
        ...state,
        isRepaying: true
      };
    case actionDescriptors.repaySuccessful:
      return {
        ...state,
        isRepaying: false
      };
    case actionDescriptors.repayFailed:
        return {
          ...state,
          error: action.error,
          isRepaying: false
        };

    case actionDescriptors.getReserve:
        return {
          ...state,
          isReserveLoading: true
        };
    case actionDescriptors.getReserveSuccessful:
        return {
          ...state,
          reserve: action.payload,
          isReserveLoading: false
        };
    case actionDescriptors.getReserveFailed:
        return {
          ...state,
          error: action.error,
          isReserveLoading: false
        };
    
    case actionDescriptors.getAllReserve:
        return {
          ...state,
          isReservesLoading: true
        };
    case actionDescriptors.getAllReserveSuccessful:
        return {
          ...state,
          reserves: action.payload,
          isReservesLoading: false
        };
    case actionDescriptors.getAllReserveFailed:
        return {
          ...state,
          error: action.error,
          isReservesLoading: false
        };
    
    case actionDescriptors.getEscrowForAsset:
        return {
          ...state,
          isEscrowLoading: true
        };
    case actionDescriptors.getEscrowForAssetSuccessful:
        return {
          ...state,
          escrow: action.payload,
          isEscrowLoading: false
        };
    case actionDescriptors.getEscrowForAssetFailed:
        return {
          ...state,
          error: action.error,
          isEscrowLoading: false
        };

    case actionDescriptors.getUserCataRewards:
        return {
          ...state,
          isUserCataRewardsLoading: true
        };
    case actionDescriptors.getUserCataRewardsSuccessful:
        return {
          ...state,
          totalCataReward: action.payload.totalCataReward,
          dailyCataReward: action.payload.dailyCataReward,
          isUserCataRewardsLoading: false
        };
    case actionDescriptors.getUserCataRewardsFailed:
        return {
          ...state,
          error: action.error,
          isUserCataRewardsLoading: false
        };
    
    case actionDescriptors.getOracle:
          return {
            ...state,
            isFetchingOracle: true
          };
    case actionDescriptors.getOracleSuccessful:
          return {
            ...state,
            oracle: action.payload,
            isFetchingOracle: false
          };
    case actionDescriptors.getOracleFailed:
          return {
            ...state,
            error: action.error,
            isFetchingOracle: false
          };    
    // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.resellInventory:
      return {
        ...state,
        isReselling: true,
      };
    case actionDescriptors.resellInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isReselling: false,
      };
    case actionDescriptors.resellInventoryFailed:
      return {
        ...state,
        error: action.error,
        isReselling: false,
      };
    case actionDescriptors.fetchSupportedTokens:
      return {
        ...state,
        isFetchingTokens: true,
      };
    case actionDescriptors.fetchSupportedTokensSuccessful:
      return {
        ...state,
        supportedTokens: action.payload,
        isFetchingTokens: false,
      };
    case actionDescriptors.fetchSupportedTokensFailed:
      return {
        ...state,
        error: action.error,
        isFetchingTokens: false,
      };
    case actionDescriptors.bridgeInventory:
      return {
        ...state,
        isBridging: true,
      };
    case actionDescriptors.bridgeInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isBridging: false,
      };
    case actionDescriptors.bridgeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isBridging: false,
      };
    case actionDescriptors.transferInventory:
      return {
        ...state,
        isTransferring: true,
      };
    case actionDescriptors.transferInventorySuccessful:
      return {
        ...state,
        isTransferring: false,
      };
    case actionDescriptors.transferInventoryFailed:
      return {
        ...state,
        error: action.error,
        isTransferring: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistory:
      return {
        ...state,
        isInventoryOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchInventoryOwnershipHistorySuccessful:
      return {
        ...state,
        inventoryOwnershipHistory: action.payload,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryDetail:
      return {
        ...state,
        isInventoryDetailsLoading: true,
      };
    case actionDescriptors.fetchInventoryDetailSuccessful:
      return {
        ...state,
        inventoryDetails: action.payload,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.fetchInventoryDetailFailed:
      return {
        ...state,
        error: action.error,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.uploadImage:
      return {
        ...state,
        isUploadImageSubmitting: true,
      };
    case actionDescriptors.uploadImageSuccessful:
      return {
        ...state,
        uploadedImg: action.payload,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.uploadImageFailed:
      return {
        ...state,
        error: action.error,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.createItem:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createItemSuccessful:
      return {
        ...state,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchPriceHistory:
      return {
        ...state,
        isFetchingPriceHistory: true,
      };
    case actionDescriptors.fetchPriceHistorySuccessful:
      return {
        ...state,
        isFetchingPriceHistory: false,
        priceHistory: action.payload,
      };
    case actionDescriptors.fetchPriceHistoryFailed:
      return {
        ...state,
        isFetchingPriceHistory: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
