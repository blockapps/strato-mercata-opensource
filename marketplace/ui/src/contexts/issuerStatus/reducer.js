import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.requestReview:
      return {
        ...state,
        requestingReview: true,
      };
    case actionDescriptors.requestReviewSuccessful:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.requestReviewFailed:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.authorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.authorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.authorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.deauthorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.modifyAdmin:
      return {
        ...state,
        changingAdminStatus: true,
      };
    case actionDescriptors.modifyAdminSuccessful:
      return {
        ...state,
        changingAdminStatus: false,
      };
    case actionDescriptors.modifyAdminFailed:
      return {
        ...state,
        changingAdminStatus: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
