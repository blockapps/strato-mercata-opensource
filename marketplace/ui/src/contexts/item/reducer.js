import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchItem:
      return {
        ...state,
        isItemsLoading: true,
      };
    case actionDescriptors.fetchItemSuccessful:
      return {
        ...state,
        items: action.payload,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchItemFailed:
      return {
        ...state,
        error: action.error,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchSerialNumbers:
      return {
        ...state,
        isSerialNumbersLoading: true,
      };
    case actionDescriptors.fetchSerialNumbersSuccessful:
      return {
        ...state,
        serialNumbers: action.payload,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchSerialNumbersFailed:
      return {
        ...state,
        error: action.error,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistory:
      return {
        ...state,
        isOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchItemOwnershipHistorySuccessful:
      return {
        ...state,
        ownershipHistory: action.payload,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemRawMaterials:
      return {
        ...state,
        isRawMaterialsLoading: true,
      };
    case actionDescriptors.fetchItemRawMaterialsSuccessful:
      return {
        ...state,
        rawMaterials: action.payload,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.fetchItemRawMaterialsFailed:
      return {
        ...state,
        error: action.error,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.setActualRawMaterials:
      return {
        ...state,
        actualRawMaterials: action.payload,
      };
    case actionDescriptors.transferOwnership:
      return {
        ...state,
        isTransferringItems: true,
      };
    case actionDescriptors.transferOwnershipSuccessful:
      return {
        ...state,
        isTransferringItems: false,
      };
    case actionDescriptors.transferOwnershipFailed:
      return {
        ...state,
        error: action.error,
        isTransferringItems: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
