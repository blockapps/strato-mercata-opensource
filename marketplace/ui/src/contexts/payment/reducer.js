import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.getPaymentServices:
      return {
        ...state,
        arePaymentServicesLoading: true,
      };
    case actionDescriptors.getPaymentServicesSuccessful:
      return {
        ...state,
        paymentServices: action.payload.data,
        paymentServicesTotal: action.payload.count,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getPaymentServicesFailed:
      return {
        ...state,
        error: action.error,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getNotOnboarded:
      return {
        ...state,
        areNotOnboardedLoading: true,
      };
    case actionDescriptors.getNotOnboardedSuccessful:
      return {
        ...state,
        notOnboarded: action.payload.data,
        notOnboardedTotal: action.payload.count,
        areNotOnboardedLoading: false,
      };
    case actionDescriptors.getNotOnboardedFailed:
      return {
        ...state,
        error: action.error,
        areNotOnboardedLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
