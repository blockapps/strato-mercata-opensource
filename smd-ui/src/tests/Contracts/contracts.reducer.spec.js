import reducer from '../../components/Contracts/contracts.reducer';
import { contracts, filter, contractsState, error, reducerContract } from "./contractsMock";
import {
  fetchContracts,
  fetchContractsSuccess,
  fetchContractsFailure,
  changeContractFilter
} from '../../components/Contracts/contracts.actions';
import {
  fetchStateSuccess, selectContractInstance, fetchAccount, fetchAccountSuccess, fetchCirrusInstancesSuccess
} from '../../components/Contracts/components/ContractCard/contractCard.actions';
import { deepClone } from '../helper/testHelper';

describe('Contracts: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('fetch contracts', () => {

    // FETCH_CONTRACTS
    test('request', () => {
      const action = fetchContracts();
      const initialState = {
        contracts: contractsState,
        filter,
        isLoading: true
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // FETCH_CONTRACTS_SUCCESSFUL
    describe('on success', () => {

      test('with initialstate of contract do not have values', () => {
        const action = fetchContractsSuccess(contracts);
        const initialState = {
          contracts: {},
          filter: '',
          error: null,
          isLoading: false
        }
        expect(reducer(initialState, action)).toMatchSnapshot();
      });

      test('with initialstate of contract has values', () => {
        const data = {
          "Cloner": [
            {
              "createdAt": 1512480630000,
              "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d"
            },
            {
              "createdAt": 1512480770000,
              "address": "2c6619f0418c2f191e2225091f7692363a91c336"
            }
          ],
          "Diff": [
            {
              "createdAt": 1512480630000,
              "address": "d07e932212f7f121b6948ffd96e1d4c726c8395d"
            },
            {
              "createdAt": 1512480770000,
              "address": "2c6619f0418c2f121e2225091f7692363a91c336"
            }
          ]
        };
        const action = fetchContractsSuccess(data);
        const initialState = {
          contracts: deepClone(contractsState),
          filter: '',
          error: null,
          isLoading: false
        }

        expect(reducer(initialState, action)).toMatchSnapshot();
      });

    });

    // FETCH_CONTRACTS_FAILED
    test('on failure', () => {
      const action = fetchContractsFailure(error);
      const initialState = {
        contracts: {},
        filter: '',
        error,
        isLoading: false
      };
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

  // CHANGE_CONTRACT_FILTER
  test('on change contract filter', () => {
    const action = changeContractFilter(filter);
    const initialState = {
      contracts: contractsState,
      error: null,
      isLoading: false
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  // FETCH_STATE_SUCCESS
  describe('fetch state', () => {

    test('update with appending state in contracts', () => {
      const action = fetchStateSuccess(reducerContract.name, reducerContract.address, reducerContract.state);
      let initialContracts = deepClone(contractsState);
      initialContracts['GreeterA']['instances'][0]['selected'] = true;
      const initialState = {
        contracts: initialContracts,
        filter: filter,
        error: 'ERROR'
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test('update without appending state in contracts', () => {
      const action = fetchStateSuccess(reducerContract.name, 'd07e932212f7f368b6948ffd96e1d4c726c8395', reducerContract.state);
      let initialContracts = deepClone(contractsState);
      initialContracts['GreeterA']['instances'][0]['selected'] = true;
      const initialState = {
        contracts: initialContracts,
        filter: filter,
        error: 'ERROR'
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

  // SELECT_CONTRACT_INSTANCE
  test('select contract instance', () => {
    const action = selectContractInstance(reducerContract.name, reducerContract.address);
    const initialState = {
      contracts: contractsState,
      filter,
      error
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  // FETCH_ACCOUNT_SUCCESS
  test('fetch account success', () => {
    const action = fetchAccountSuccess(reducerContract.name, reducerContract.address, reducerContract.account);
    let initialContracts = deepClone(contractsState);
    initialContracts['GreeterA']['instances'][0]['selected'] = true;
    const initialState = {
      contracts: initialContracts,
      filter
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  // FETCH_CIRRUS_INSTANCES_SUCCESS
  test('fetch cirrus instances success', () => {
    const data = {
      instances: [
        {
          address: "b7b986bf23faebd8d745c65fa42a8c2f0fc2ebb9",
          greetingB: ""
        }],
      name: "GreeterB"
    }
    const action = fetchCirrusInstancesSuccess(data.name, data.instances);
    const initialState = {
      contracts: contractsState,
      filter,
      error,
      isLoading: false
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

})
