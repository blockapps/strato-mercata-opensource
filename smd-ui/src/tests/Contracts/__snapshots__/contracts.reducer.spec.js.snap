// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contracts: reducer fetch account success 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "balance": 0,
          "createdAt": 1512481078000,
          "fromBloc": true,
          "selected": true,
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": null,
  "filter": "Just to test filter",
}
`;

exports[`Contracts: reducer fetch cirrus instances success 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
          "fromCirrus": false,
        },
        Object {
          "address": "b7b986bf23faebd8d745c65fa42a8c2f0fc2ebb9",
          "fromCirrus": true,
          "greetingB": "",
        },
      ],
    },
  },
  "error": "ERROR",
  "filter": "Just to test filter",
}
`;

exports[`Contracts: reducer fetch contracts on failure 1`] = `
Object {
  "contracts": Object {},
  "error": "ERROR",
  "filter": "",
  "isLoading": false,
}
`;

exports[`Contracts: reducer fetch contracts on success with initialstate of contract do not have values 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": null,
  "filter": "",
  "isLoading": false,
}
`;

exports[`Contracts: reducer fetch contracts on success with initialstate of contract has values 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "Diff": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f121b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f121e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": null,
  "filter": "",
  "isLoading": false,
}
`;

exports[`Contracts: reducer fetch contracts request 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": null,
  "filter": "Just to test filter",
  "isLoading": true,
}
`;

exports[`Contracts: reducer fetch state update with appending state in contracts 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "createdAt": 1512481078000,
          "fromBloc": true,
          "selected": true,
          "state": Object {
            "greetA": "function () returns (Address)",
            "greetingA": "Aaaaaaaaa",
          },
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": "ERROR",
  "filter": "Just to test filter",
}
`;

exports[`Contracts: reducer fetch state update without appending state in contracts 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "createdAt": 1512481078000,
          "fromBloc": true,
          "selected": true,
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": "ERROR",
  "filter": "Just to test filter",
}
`;

exports[`Contracts: reducer on change contract filter 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": null,
  "filter": "Just to test filter",
}
`;

exports[`Contracts: reducer select contract instance 1`] = `
Object {
  "contracts": Object {
    "Cloner": Object {
      "instances": Array [
        Object {
          "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
          "createdAt": 1512480630000,
          "fromBloc": true,
        },
        Object {
          "address": "2c6619f0418c2f191e2225091f7692363a91c336",
          "createdAt": 1512480770000,
          "fromBloc": true,
        },
      ],
    },
    "GreeterA": Object {
      "instances": Array [
        Object {
          "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
          "createdAt": 1512481078000,
          "fromBloc": true,
          "selected": true,
        },
      ],
    },
    "GreeterB": Object {
      "instances": Array [
        Object {
          "address": "GreeterA",
          "createdAt": 1512481078000,
          "fromBloc": true,
        },
      ],
    },
  },
  "error": "ERROR",
  "filter": "Just to test filter",
}
`;

exports[`Contracts: reducer set initial state 1`] = `
Object {
  "contracts": Object {},
  "error": null,
  "filter": "",
  "isLoading": false,
}
`;
